Support gamma on CUDA.

plietar opened the pull request:
This worked out of the box simply by adding the right annotations. I've tested this on all three parts of the domain (`shape < 1`, `shape == 1` and `shape > 1`) and get results that are equivalent on the CPU and GPU (given the same starting seed).

Oddly enough I do get very occasional insignificant floating point differences between the two, despite using doubles on both. I suspect this is down to differences in the some of the math functions. By comparison, I get no difference at all on the uniform random number generator.

richfitz committed:
Avoid writing to /dev/null

This has been cherry-picked from dust2:
https://github.com/mrc-ide/dust2/commit/0312a663828217b895b3216f1fda8b9165e49a94

plietar committed:
Support gamma on CUDA.

This worked out of the box simply by adding the right annotations. I've
tested this on all three parts of the domain (`shape < 1`, `shape == 1`
and `shape > 1`) and get results that are equivalent on the CPU and GPU
(given the same starting seed).

Oddly enough I do get very occasional insignificant floating point
differences between the two, despite using doubles on both. I suspect
this is down to differences in the some of the math functions. By
comparison, I get no difference at all on the uniform random number
generator.

plietar committed:
Fix version number.

I was told to do this by the CI. Not sure how it got out of sync with
DESCRIPTION.

I had to tweak the call to the glue package a bit as the `glue::glue`
function only accepts environments now, not named lists. See
https://github.com/tidyverse/glue/commit/963ea6a43d21c2224b2d5a6747e409abf48e4443

richfitz reviewed (changes requested):
I've seen these minor differences before, that was part of the beginning of the end of hoping we could get identical results on the cpu and gpu based code.  Can you bump the artefact action version on those failing actions - I think they'll run ok after that

plietar committed:
update artifact action

plietar committed:
Don't use sudo to install Python packages.

When using `sudo pip3 install`, it will try to install the packages
globally which fails because apt has already installed its own copy of
pygments.

Just running it as the normal user seems to work.

plietar committed:
Fix docker build.

Update rocker image, add missing dependencies.
